src/popper/runner_kubernetes.py

```diff
@@ -110,10 +110,19 @@ def _init_pod_schedule(self):
             pod_host_node = self._config.resman_opts.pod_host_node
 
         elif not self._config.resman_opts.get("persistent_volume_name", None):
-            nodes = [
-                node.metadata.labels["kubernetes.io/hostname"]
-                for node in self._kclient.list_node().items
-            ]
+            nodes = []
+
+            for node in self._kclient.list_node().items:
+                node_role = ""
+                if node.spec.taints is not None:
+                    node_role = node.specs.taints.key
```

Comment:
It will be `node.spec` not `node.specs`.

Reply:
Also, node.spec.taints is a list. We need to just check the first element of the taints list. the first element has the information about if the node whether it is a master, or is unreachable.

Reply:
Ah. Sorry for missing those typos. 

As for taints, are you saying `node.spec.taints[0]` would be better as opposed to `node.spec.taints.key?` And is the `is not None` line needed? I was uncertain if you meant taints could be null or simply empty on slack. 

Thank you very much for the continued help and time.

Reply:
spec may or may not have the taints key. if the taints key , the value it will have is a list. Now this taints list can either be empty or non-empty. if non-empty, pick the first element and check.